Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -239,19 +242,37 @@ public async Task<IEnumerable<SubmissionForProfileServiceModel>> GetForProfileBy
         return await userSubmissions.ToListAsync();
     }
 
+    public async Task<IEnumerable<SubmissionResultsServiceModel>> GetSubmissionResultsByProblemAndUser(int problemId,
+        bool isOfficial, string userId, int take = 0)
+    {
+        var problem = await this.problemsDataService.GetWithProblemGroupById(problemId);
+
+        await this.ValidateUserCanViewResults(problem, isOfficial);
+
+       var userSubmissions = this.submissionsData
+                .GetAllByProblemAndUser(problemId, userId)
+                .MapCollection<SubmissionResultsServiceModel>();
+
+        if (take != 0)
+        {
+            userSubmissions = userSubmissions.Take(take);
+        }
```

Comment:
Well, no.
This IS strange. It's called `hidden` behavior. If I provide `0` as `take` I would expect to get an empty result, not all of them. 

Make it `nullable`. `null` is better suited for that.